docs/reference/ois/next/processing.md

```diff
@@ -77,14 +93,84 @@ post-processing.
 
 :::
 
+### v2
+
+The processing snippet receives parameters submitted as part of a template or
+on-chain request.
+
+Airnode executes snippets for `preProcessingSpecificationV2` and
+`postProcessingSpecificationV2` during its run cycle. The following describes
+the work flow Airnode uses:
+
+1. Run `preProcessingSpecificationV2`
+2. Airnode calls requested OIS endpoint
+3. Run `postProcessingSpecificationV2`
+4. Airnode encodes the response values defined by reservedParameters
+
+The processing schema is the same for both
+[`preProcessingSpecificationV2`](/reference/ois/next/specification.md#_5-11-preprocessingspecificationv2)
+and
+[`postProcessingSpecificationV2`](/reference/ois/next/specification.md#_5-12-postprocessingspecificationv2).
+Snippets for both specifications follow this schema:
+
+- `environment` - Currently only possible value is `Node`. This options
+  interprets the code as JavaScript and execute in Node.js. The function can be
+  also asynchronous (async/await is supported as well). The processing
+  implementation will wait for the function to resolve.
+- `value` - The processing code written as a string.
+- `timeoutMs` - The maximum timeout that this snippet can run. In case the
+  timeout is exceeded an error is thrown.
+
+Try the [Post processing](/guides/airnode/post-processing/) guide to further
+understand pre/post processing.
+
+#### Input and Output
+
+The processing snippet is a function which receives payload as an argument. The
+return value of the function is treated as a processing result. Apart from the
+payload argument, you can use most Node.js built-in modules.
+
+The payload argument for pre-processing is an object with the following
+properties:
+
+- `apiCallParameters` - The API call parameters with the exception of reserved
+  parameters. For example, if there was a parameter named `myParameter` defined
+  in the `endpoints[n].parameters` array, its value could be accessed using
+  `endpointParameters.myParameter` within pre-processing snippet.
+
+The output of the pre-processing snippet is an object with the following fields:
+
+- `apiCallParameters` - The pre-processed API call parameters. These are used to
+  make the API call.
+
+The payload argument for post-processing is an object with the following
+properties:
+
+- `apiCallResponse` - The API call response.
+- `endpointParameters` - The API call parameters with the exception of reserved
+  parameters. For example, if there was a parameter named `myParameter` defined
+  in the `endpoints[n].parameters` array, its value could be accessed using
+  `endpointParameters.myParameter` within pre-processing snippet.
+
+The output of the post-processing snippet is an object with the following
+fields:
+
+- `apiCallResponse` - The post-processed API call response. This is used to
```

Comment:
This is not the API call response, it's the post-processed version of it, so this feels misnamed to me

Reply:
My reasoning is that you can return the same response that you receive in the function, similarly to doing `output = input` in the v1 processing.

Reply:
In v1, `output` is named differently than `input`. The equivalent of `output = input` would be `apiCallResponse = postProcessedApiCallResponse`.

Reply:
It would actually be `postProcessedApiCallResponse  = apiCallResponse`, but since we have function what you want is `return apiCallResponse` but we can't do that, because we also want to (optionally) return timestamp so we need to do it like `return {apiCallResponse}`.

Reply:
I was thinking a bit more about:

> This is not the API call response, it's the post-processed version of it, so this feels misnamed to me

Since Airnode/Pusher does the processing and its deployed by the API provider themselves, to me the API response is actually a combination of API call + post-processing. However as you've said, there is possibility of skipping API call in which case the apiCallResponse (or apiResponse) don't make sense and I can't think of a good name for this.

One possibility I had in mind would be to call it `apiCallResponse` (or `apiResponse`) and when we skip a call we would pass `undefined`, which kinda makes sense. Currently, we [use the pre-processing output as API response](https://github.com/api3dao/airnode/blob/02d15254dfc431b4c0aa06b36f64953c550849fd/packages/airnode-node/src/api/index.ts#L323) so with this change the post-processing would be a must have when skipping an API call.

Reply:
The more technically correct output might be `postProcessedApiCallResponse`, but I don't disagree with this either:
> to me the API response is actually a combination of API call + post-processing

Post-processing as an "inplace" update to the API call response is effectively what we're doing i.e. don't use the actual API call response, use this instead.

Having a different return variable would also mean that modifying only the timestamp would eliminate property value shorthand:

```ts
"async ({ apiCallResponse }) => { return { postProcessedApiCallResponse: apiCallResponse, timestamp: apiCallResponse.ProviderTimestamp }; }"
```

tl;dr I'd be fine leaving it as `apiCallResponse`.

Reply:
I personally see pre/post-processing as a part of the endpoint specs and not API functionality. However, leaving this as `apiCallResponse` would be consistent with having the output of pre-processing be named `endpointParameters` (which I had agreed on above).
One final nit, one can use post-processing without an API call (https://docs.api3.org/reference/ois/latest/processing.html#skip-the-api-call), in which case this naming would be off. The previous `input`/`output` names worked nicely with this kind of ambiguity. An easy fix would be to call this `response` instead of `apiCallResponse`.

Reply:
Calling it `response` and clarifying this in the docs.


As a last thing, @dcroote @bbenligiray What do you think about:
> One possibility I had in mind would be to call it apiCallResponse (or apiResponse) and when we skip a call we would pass undefined, which kinda makes sense. Currently, we [use the pre-processing output as API response](https://github.com/api3dao/airnode/blob/02d15254dfc431b4c0aa06b36f64953c550849fd/packages/airnode-node/src/api/index.ts#L323) so with this change the post-processing would be a must have when skipping an API call.

Reply:
Then pre-processing (v2) while the API call is skipped is entirely useless so it shouldn't be allowed imo.
Also, you should disallow using different versions (v1/v2) of pre and post processing unless you haven't already.

Reply:
For v1 we allow either pre or post-processing when an API call is skipped: https://docs.api3.org/reference/ois/latest/processing.html#skip-the-api-call
>Instead of calling an API, Airnode uses the output of preProcessingSpecifications, postProcessingSpecifications, or both.


But I agree with Burak that pre doesn't really make sense if the API call is skipped.

Reply:
Ok. Resolving the discussion and going with the current state. I think everything should now be as we agreed.